### ISSUE TITLE
Node 18; Func v4; VSCode on macOS: "No job functions found. Try making your job classes and methods public."

### ISSUE THREAD

**Issue by guice** [Azure/azure-functions-nodejs-worker#713]
I'm trying to get VSCode to run the new v4 functions locally, but I keep getting function not found error. I've followed the [Troubleshooting](https://learn.microsoft.com/en-us/azure/azure-functions/functions-node-upgrade-v4#troubleshoot) steps, but that only affected _remote_ settings. Remote is working. **Locally is NOT** (this example, I use cli directly. I also used F5 with the same results):

![Screenshot 2023-08-23 at 3 29 51 PM](https://github.com/Azure/azure-functions-nodejs-worker/assets/205378/d18b286f-e6f5-4efe-b349-38ffa816aa6b)

I stumbled upon this issue https://github.com/Azure/azure-functions-nodejs-worker/issues/675 which said to open another issue. So, I am. I'm trying to give as much detail as I can here.

What am I missing?

The code is right: it works in Azure. It's just not working when I hit `F5`.

Node: 18.17.1
Functions: 4.0.5274
VSCode Env: 
> Version: 1.81.1
> Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
> Date: 2023-08-09T22:40:25.698Z
> Electron: 22.3.18
> ElectronBuildId: 22689846
> Chromium: 108.0.5359.215
> Node.js: 16.17.1
> V8: 10.8.168.25-electron.0
> OS: Darwin arm64 22.6.0

Azure Functions plugin: v1.12.4
 > Project Runtime: ~4

Code does work on Azure, but here's my package.json anyway so you can see I am use v4 tools:
```json
{
  "name": "b2c-token-manager",
  "version": "1.0.0",
  "main": "src/functions/*.js",
  "description": "",
  "scripts": {
    "start": "func start",
    "test": "echo \"No tests yet...\""
  },
  "dependencies": {
    "@azure/functions": "^4.0.0-alpha.11"
  },
  "devDependencies": {
    "azure-functions-core-tools": "^4.x"
  }
}
```

**Comment by ejizba** [comment]
Hi @guice, can you double check you have `"AzureWebJobsFeatureFlags": "EnableWorkerIndexing"` in your `local.settings.json`?

If that doesn't help, please add `"logging__logLevel__Worker": "information"` to your local settings and run your app again to get more detailed logs. You can either check the logs yourself or post them here

**Comment by guice** [comment]
> can you double check you have "AzureWebJobsFeatureFlags": "EnableWorkerIndexing" in your local.settings.json?

That was it! Oh geez, this is why. I missed this line:

> When you're running locally, add AzureWebJobsFeatureFlags with a value of EnableWorkerIndexing to your _local.settings.json_ file

Thanks. It's working now.

**Comment by guice** [comment]
> can you double check you have "AzureWebJobsFeatureFlags": "EnableWorkerIndexing" in your local.settings.json?

That was it! Oh geez, this is why. I missed this line:

> When you're running locally, add AzureWebJobsFeatureFlags with a value of EnableWorkerIndexing to your _local.settings.json_ file

Thanks. It's working now. I'll close the PR now. It's resolved.

**Comment by garrytrinder** [comment]
Adding this comment for folks who run into this issue when migrating from v3 to v4. Only after adding `"AzureWebJobsFeatureFlags": "EnableWorkerIndexing"` to my `local.settings.json` file was I able to get my HTTP Trigger functions working locally.
